test/core/core_online_test.cpp

```diff
@@ -83,6 +85,8 @@ TestCoreOnline::TestCoreOnline()
     qRegisterMetaType<ToxPk>("ToxPk");
     qRegisterMetaType<uint32_t>("uint32_t");
     qRegisterMetaType<Status::Status>("Status::Status");
+    auto timeout = QString("%1").arg(very_long_timeout);
+    assert(qputenv("QTEST_FUNCTION_TIMEOUT", timeout.toLatin1()));
```

Comment:
This will get removed if you try to run your unit tests with optimizations, so doing things with side-effects inside `assert` is a bit weird. :x

Reply:
Crap missed this comment before merging. That's a good point, will fix.